client/extensions/woocommerce/woocommerce-services/components/dropdown/index.js

```diff
@@ -26,7 +26,7 @@ const Dropdown = ( {
 	disabled,
 	className,
 } ) => {
-	const onChange = event => updateValue( event.target.value );
+	const onChange = useCallback(event => updateValue( event.target.value, event ), [ updateValue ]);
```

Comment:
- Wrapped the `onChange` with a `useCallback` to reduce unwanted re-renders.
- Passing the original `event` as the second argument to the `updateValue` prop, so it can be leveraged by consumers

Reply:
In order to test the re-renders you'd have to use the browser's devtools, but you need to be wary to the usages.

If a consumer of `Dropdown` is written as follows:
```
const ConsumerOfDropdown = ({someOtherProp}) => {
  const [value, setValue] = useState('');

  // updateValue is not memoized, causing re-renders every time `ConsumerOfDropdown` is rendered (every time its props or internal state changes
  return <Dropdown updateValue={(updatedValue) => setValue(updatedValue)} value={value} />
}
```

Then there wouldn't be a benefit, because when the props or the state change, also the `Dropdown` tree is going to re-render.

But if the consumer is instead written as:
```
const ConsumerOfDropdown = ({someOtherProp}) => {
  const [value, setValue] = useState('');

  // updateValue is now memoized
  const handleChange = useCallback((updatedValue) => setValue(updatedValue), [setValue]);

  return <Dropdown updateValue={handleChange} value={value} />
}
```

Then the `Dropdown` component can now re-render only when `updateValue` or `value` change, not when `someOtherProp` does.

One way to test it is to have the React extension on your browser and look at the flamegraph generated by the "Profiling" tab, one you hit "record"
![Screen Shot 2020-12-01 at 3 23 08 PM](https://user-images.githubusercontent.com/273592/100798820-c32f8500-33e9-11eb-8961-9dc53e32c64a.png)
It can get hard to read, especially when there's a lot of components interacting with each other. But that's a good tool to identify possibly unwanted re-renders.
But again, you'd need to be wary of the usages of the component :) In this specific case I'm not sure there'd be a noticeable difference